### ISSUE TITLE
Feature request: Silence not-found error when a suggested import is already in-place but typo'd

### ISSUE THREAD

**Issue by jplatte** [rust-lang/rust#84093]
When trying to import and then use an item, but there is a typo in the import (which rustc successfully provides a suggestion for), I think the usage of the item should not trigger a not-found error. Example with a trait method [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a524258075f42fbcb532cc7017b0a1cb):

```rust
use std::convert::Tryinto;

fn main() {
    let x = 0i32;
    let y: usize = x.try_into();
}
```

Currently triggers two errors, with the second one being redundant IMO:

```
error[E0432]: unresolved import `std::convert::Tryinto`
 --> src/lib.rs:1:5
  |
1 | use std::convert::Tryinto;
  |     ^^^^^^^^^^^^^^-------
  |     |             |
  |     |             help: a similar name exists in the module (notice the capitalization): `TryInto`
  |     no `Tryinto` in `convert`

error[E0599]: no method named `try_into` found for type `i32` in the current scope
 --> src/lib.rs:5:22
  |
5 |     let y: usize = x.try_into();
  |                      ^^^^^^^^ method not found in `i32`
  |
  = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
  |
1 | use std::convert::TryInto;
  |
```
